src/App.jsx

```diff
@@ -1,10 +1,20 @@
 import "./App.css";
 import TodoList from "./TodoList.jsx";
 import AddTodoForm from "./AddTodoForm.jsx";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(
+    JSON.parse(localStorage.getItem("savedTodoList"))
```

Comment:
because you are getting the `savedTodoList` from local storage, this will crash if there is nothing on your local storage so you need to provide a fallback

```suggestion
    JSON.parse(localStorage.getItem("savedTodoList")) || []
```